packages/slate/src/changes/at-range.js

```diff
@@ -863,8 +863,8 @@ Changes.insertTextAtRange = (change, range, text, marks, options = {}) => {
   }
 
   // PERF: Unless specified, don't normalize if only inserting text.
-  if (normalize !== undefined) {
-    normalize = range.isExpanded
+  if (normalize === undefined) {
+    normalize = range.isExpanded && marks.size !== 0
```

Comment:
marks needs a null-guard, given its not even in the docs. https://docs.slatejs.org/slate-core/change#inserttextatrange